docs/rfcs/2025-08-16-async-index-build.md

```diff
@@ -0,0 +1,110 @@
+---
+Feature Name: Async Index Build
+Tracking Issue: TBD
+Date: 2025-08-16
+Author: "SNC123 <sinhco@outlook.com>"
+---
+
+# Summary
+This RFC proposes an asynchronous index build mechanism in the database, with a configuration option to choose between synchronous and asynchronous modes, aiming to improve flexibility and adapt to different workload requirements.
+
+# Motivation
+Currently, index creation is performed synchronously, which may lead to prolonged write suspension and impact business continuity. As data volume grows, the time required for index building increases significantly. An asynchronous solution is urgently needed to enhance user experience and system throughput.
+
+# Details
+
+## Overview
+
+The following table highlights the difference between async and sync index approach:
+
+| Approach | Trigger | Data Source | Index Metadata Update | Fine-grained `FileMeta` Index |
```

Comment:
I think the `Index Metadata Update`  isn't `Not Required` but executed synchronously?